Scripts/datatypes/tour.py

```diff
@@ -179,6 +182,54 @@ def choose_secondary_destination(self, cumulative_probs):
                     + numpy.searchsorted(cumulative_probs, self._sec_dest_draw))
         self.position = (self.position[0], self.position[1], dest_idx)
         self.purpose.sec_dest_purpose.attracted_tours[self.mode][dest_idx] += 1
+    
+    def calc_cost(self, impedance):
+        """Construct cost and time components from tour dest choice.
+
+        Parameters
+        ----------
+        impedance: dict
+            Time period (aht/pt/iht) : dict
+                Type (time/cost/dist) : dict
+                    Assignment class (car_work/transit/...) : numpy 2d matrix
+        """
+        time = self._get_cost(impedance, "time")
+        self.cost = self._get_cost(impedance, "cost")
+        vot = 1 / vot_inv[assignment_classes[self.purpose_name]]
+        self.gen_cost = self.cost + time * vot
+
+    def _get_cost(self, impedance, mtx_type):
+        """Get cost and time components from tour dest choice."""
+        if self.mode in divided_classes:
+            ass_class = "{}_{}".format(
+                self.mode, assignment_classes[self.purpose.name])
+        else:
+            ass_class = self.mode
+        cost = 0
+        try:
+            if assignment_classes[self.purpose_name] == "work":
+                departure_imp = impedance["aht"][mtx_type][ass_class]
+                sec_dest_imp = impedance["iht"][mtx_type][ass_class]
+                return_imp = impedance["iht"][mtx_type][ass_class]
+            else:
+                departure_imp = impedance["pt"][mtx_type][ass_class]
+                sec_dest_imp = impedance["pt"][mtx_type][ass_class]
+                return_imp = impedance["pt"][mtx_type][ass_class]
+            # first leg of tour
+            cost += departure_imp[self.position[0], self.position[1]]
+            # check if tour has secondary destination and add accordingly
+            if len(self.position) > 2:
+                cost += sec_dest_imp[self.position[1], self.position[2]]
+                cost += return_imp[self.position[2], self.position[0]]
+            else: 
+                cost += return_imp[self.position[1], self.position[0]]
+        except KeyError:
+            pass
```

Comment:
I think this try-except clause could be considered to wide, we risk excepting unintended bugs. But I cannot come up with a more elegant solution.

Reply:
Maybe we could add a comment describing what the except clause is supposed to catch?